src/crt.c

```diff
@@ -135,9 +135,13 @@ static void resource_dtor(void *ptr) {
     aws_crt_mem_release(ptr);
 }
 
+void aws_crt_resource_init(aws_crt_resource *resource) {
+    aws_ref_count_init(&resource->rc, resource, resource_dtor);
+}
+
 void *aws_crt_resource_new(void *object) {
     aws_crt_resource *resource = object;
-    aws_ref_count_init(&resource->rc, resource, resource_dtor);
+    aws_crt_resource_init(resource);
     return object;
 }
```

Comment:
Now that you have 2 of these functions. the new() one is just weird. It's named _new() but it doesn't allocate. But I think you named it new because you're always using it like this???
`aws_crt_event_loop_group *elg = aws_crt_resource_new(aws_crt_mem_calloc(1, sizeof(aws_crt_event_loop_group)));`

Maybe you should do 1 of these things:
1) just have the _init() function and no new() function, since this isn't actually allocating
2) change the _new() function api to be like:
`void *aws_crt_resource_new(size_t sizeof_object); /* don't worry I'll call calloc for you */`
and then you can use it like:
`aws_crt_event_loop_group *elg = aws_crt_resource_new(sizeof(aws_crt_event_loop_group));`

Reply:
I like 2. That's a good factoring of what was already happening.